libs/chain-signatures/contract/src/lib.rs

```diff
@@ -823,56 +876,88 @@ impl VersionedMpcContract {
         }
     }
 
+    /// Upon success, removes the signature from state and returns it.
+    /// Returns an Error if the signature timed out.
+    /// Note that timed out signatures will need to be cleaned up from the state by a different function.
+    #[private]
+    #[handle_result]
+    pub fn return_signature_and_clean_state_on_success(
+        &mut self,
+        contract_signature_request: ContractSignatureRequest,
```

Comment:
Wait so why is it that we cannot do the cleanup here in the timeout case? This argument is the only one needed for the cleanup.

Reply:
If my assumption is correct, we could still do the cleanup here, but would need to return `Ok(<SignatuseResult, Error>)` and then have the caller handle it. But I think this would break backwards-compatibility.

Reply:
Ohh, I see, thanks!